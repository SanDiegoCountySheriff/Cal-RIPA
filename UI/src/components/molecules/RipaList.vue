<template>
  <div :class="[getMarginTop, getMarginLeft]">
    <template v-if="item.level === 1">
      {{ item.header }}
      <span class="tw-ml-2" :style="{ color: getPrimaryColor }">{{
        item.detail
      }}</span>
    </template>

    <template v-if="item.level === 2">
      <div>
        {{ item.header }}
      </div>
      <div class="tw-ml-8" v-for="child in item.children" :key="child.detail">
        <div :class="getChildMarginLeft(child)">
          <span :style="{ color: getPrimaryColor }">
            {{ child.detail }}
          </span>
        </div>
      </div>
    </template>

    <template v-if="item.level === 3">
      <div>
        {{ item.header }}
      </div>
      <div class="tw-ml-8" v-for="child in item.children" :key="child.header">
        {{ child.header }}
        <span class="tw-ml-2" :style="{ color: getPrimaryColor }">
          {{ child.detail }}
        </span>
      </div>
    </template>
  </div>
</template>

<script>
export default {
  name: 'ripa-list',

  computed: {
    getPrimaryColor() {
      return this.$vuetify.theme.themes.dark.primary
    },

    getMarginTop() {
      return this.item.marginTop ? 'tw-mt-4' : ''
    },

    getMarginLeft() {
      return this.item.marginLeft ? 'tw-ml-4' : ''
    },
  },

  methods: {
    getChildMarginLeft(child) {
      return child.marginLeft ? 'tw-ml-4' : ''
    },
  },

  props: {
    item: {
      type: Object,
      default: () => {},
    },
  },
}
</script>
