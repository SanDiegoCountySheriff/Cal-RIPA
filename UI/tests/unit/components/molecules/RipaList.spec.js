import RipaList from '@/components/molecules/RipaList.vue'
import { shallowMount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa List', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  afterEach(() => {
    wrapper.destroy()
  })

  const factory = propsData => {
    return shallowMount(RipaList, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  const item = {
    level: 1,
  }

  it('should match snapshot', () => {
    wrapper = factory({ item: item })

    expect(wrapper.element).toMatchSnapshot()
  })
})
