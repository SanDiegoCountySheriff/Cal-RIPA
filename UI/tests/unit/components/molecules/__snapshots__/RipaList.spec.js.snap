// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa List should match snapshot 1`] = `
<div class="">

  <span class="tw-ml-2" style="color: rgb(33, 150, 243);"></span>
  <!---->
  <!---->
</div>
`;
